signac/contrib/project.py

```diff
@@ -1526,11 +1518,42 @@ def init_project(cls, name, root=None, workspace=None, make_dir=True):
             configuration.
 
         """
+        if args:
+            if len(args) == 1:
+                warnings.warn(
+                    "Project names were removed in signac 2.0. If you intended to call "
+                    "`init_project` with a root directory, please provided it as a keyword "
+                    f"argument: `init_project(root={args[0]})`. If your project name "
+                    "contains important information, consider storing it in the project document "
+                    "instead.",
+                    FutureWarning,
+                )
+            else:
+                # Match the usual error from misusing keyword-only args.
+                num_args = len(args)
+                raise TypeError(
+                    f"init_project() takes 0 positional arguments but {num_args} "
+                    f"{'was' if num_args == 1 else 'were'} given"
```

Comment:
We've explicitly ruled out `num_args == 1` from the previous `if` block. Since this entire error block should be removed in the future, I would mark it as `# TODO: Remove this error when removing support for single positional argument project name` and always use "were" in this message.